src/isyntax/isyntax.c

```diff
@@ -317,29 +397,12 @@ static bool isyntax_parse_scannedimage_child_node(isyntax_t* isyntax, u32 group,
 				case 0x1005: { /*PIM_DP_IMAGE_DATA*/
 					size_t decoded_capacity = value_len;
 					size_t decoded_len = 0;
-					i32 last_char = value[value_len-1];
+					char last_char = value[value_len-1];
 					if (last_char == '/') {
 						value_len--; // The last character may cause the base64 decoding to fail if invalid
 					}
-					u8* decoded = base64_decode((u8*)value, value_len, &decoded_len);
-					if (decoded) {
-						i32 channels_in_file = 0;
-#ifdef ISYNTAX_JPEG_DECODER_USE_LIBJPEG
-						// TODO: Why does this crash?
-						// Apparently, there is a bug in the libjpeg-turbo implementation of jsimd_can_h2v2_fancy_upsample() when using SIMD.
-						// jsimd_h2v2_fancy_upsample_avx2 writes memory out of bounds.
-						// This causes the program to crash eventually when trying to free memory in free_pool().
-						// When using a hardware watchpoint on the corrupted memory, the overwiting occurs in x86_64/jdsample-avx2.asm at line 358:
-						//     vmovdqu     YMMWORD [rdi+3*SIZEOF_YMMWORD], ymm6
-						// WORKAROUND: disabled SIMD in jsimd_can_h2v2_fancy_upsample().
-
-						image->pixels = jpeg_decode_image(decoded, decoded_len, &image->width, &image->height, &channels_in_file);
-#else
-						// stb_image.h
-						image->pixels = stbi_load_from_memory(decoded, decoded_len, &image->width, &image->height, &channels_in_file, 4);
-#endif
-						free(decoded);
-					}
+					image->jpeg_compressed_pixels = base64_decode((u8*)value, value_len, &decoded_len);
```

Comment:
One note here - in my ML OpenSlide branch I disabled this, because here we both do work and consume memory, and that image may never be requested (typical for ML). Since we're touching this area, maybe worth thinking about how to make it lazy - so that we `base64_decode` and allocate memory only if/when the label/macro images were requested.

Your call - if you prefer to leave this optimization to a separate PR (maybe by me later on), I have no objections. It is no worse than what is in main now.

Reply:
Yes fair point. One caveat: if we don't at least save the base64-encoded string somewhere (by allocating memory for it) we won't be able to easily retrieve it later unless we parse the XML header again (the XML tags are not stored after looking at them once). Because the base64-encoded string takes more memory than the binary equivalent, in my mind it made sense to at least do the base64 decoding while we have the chance. If your preference for the trade-off would be to save the CPU cycles at the cost of some memory, then that's also fine for me.

Reply:
Would it be possible to save file offsets where the base64 string begins and ends? I didn't look too much into the xml parsing, so I don't really know if easy or hard to do.

But if possible. then we can re-read that section on-demand without needing to re-parse the whole file.